Bugsnag/Storage/BSGFileLocations.m

```diff
@@ -19,39 +22,44 @@ static BOOL ensureDirExists(NSString *path) {
     return YES;
 }
 
-static NSString *rootDirectory(NSString *fsVersion) {
+static NSString *rootDirectory(NSString *fsVersion, NSString *subdirectory) {
     // Default to an unusable location that will always fail.
     static NSString* defaultRootPath = @"/";
     static NSString* rootPath = @"/";
-
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
+    
 #if TARGET_OS_TV
-        // On tvOS, locations outside the caches directory are not writable, so fall back to using that.
-        // https://developer.apple.com/library/archive/documentation/General/Conceptual/AppleTV_PG/index.html#//apple_ref/doc/uid/TP40015241
-        NSSearchPathDirectory directory = NSCachesDirectory;
+    // On tvOS, locations outside the caches directory are not writable, so fall back to using that.
+    // https://developer.apple.com/library/archive/documentation/General/Conceptual/AppleTV_PG/index.html#//apple_ref/doc/uid/TP40015241
+    NSSearchPathDirectory directory = NSCachesDirectory;
 #else
-        NSSearchPathDirectory directory = NSApplicationSupportDirectory;
+    NSSearchPathDirectory directory = NSApplicationSupportDirectory;
 #endif
-        NSError *error = nil;
-        NSURL *url = [NSFileManager.defaultManager URLForDirectory:directory inDomain:NSUserDomainMask appropriateForURL:nil create:YES error:&error];
-        if (!url) {
-            bsg_log_err(@"Could not locate directory for storage: %@", error);
-            return;
-        }
-
+    NSError *error = nil;
+    NSURL *url = [NSFileManager.defaultManager URLForDirectory:directory inDomain:NSUserDomainMask appropriateForURL:nil create:YES error:&error];
+    if (!url) {
+        bsg_log_err(@"Could not locate directory for storage: %@", error);
+        return rootPath;
+    }
+    
+    if (subdirectory != nil) {
+        rootPath = [NSString stringWithFormat:@"%@/com.bugsnag.Bugsnag/%@/%@/%@/%@",
```

Comment:
Keeping path constructing style that is used in the existing code.

Reply:
You could keep the original code. If `subdirectory != nil`, append the subdirectory specific things?